src/components/OSCALLoader.js

```diff
@@ -219,7 +245,11 @@ export default function OSCALLoader(props) {
         "",
         "",
         `/${props.oscalObjectType.jsonRootName}/${
-          props.isRestMode ? `${oscalObjectUuid}` : ""
+          props.isRestMode
+            ? `${oscalObjectUuid}${
+                props.urlFragment ? `#${props.urlFragment}` : ""
+              }`
```

Comment:
This is getting very nasty. Please calculate this value _before_ the `window.history.pushState` call and then pass that variable here.

```
const fragment = props.urlFragment ? `#${props.urlFragment}` : "";
const document = props.isRestMode ? oscalObjectUuid : "";
const path = `/${props.oscalObjectType.jsonRootName}/${document}${fragment}`;